packages/form-control/src/FormControlMixin.ts

```diff
@@ -0,0 +1,528 @@
+     */
+    static getValidator(attribute: string): Validator {
+      return this.formControlValidators.find(validator =>
+        validator.attribute === attribute
+      );
+    }
+
+    /** The ElementInternals instance for the control. */
+    internals = this.attachInternals() as unknown as IElementInternals;
+
+    /** Keep track of if the control has focus */
+    focused = false;
+
+    /**
+     * Toggles to true whenever the element has been focused. This property
+     * will reset whenever the control's formResetCallback is called.
+     */
+     get touched() {
+      return this.hasAttribute('touched');
+    }
+
+    set touched(touched) {
+      this.toggleAttribute('touched', !!touched);
+    }
+
+    /**
+     * The element that will receive focus when the control's validity
+     * state is reported either by a form submission or via API
+     */
+    validationTarget: HTMLElement;
+
+    /**
+     * The controls' form value. As this property is updated, the form value
+     * will be updated. If a given control has a `checked` property, the value
+     * will only be set if `checked` is truthy.
+     */
+    value: any = '';
+
+    /** Return a reference to the control's form */
+    get form(): HTMLFormElement {
+      return this.internals.form;
+    }
+
+    /**
+     * Will return true if it is recommended that the control shows an internal
+     * error. If using this property, it is wise to listen for 'invalid' events
+     * on the element host and call preventDefault on the event. Doing this will
+     * prevent browsers from showing a validation popup.
+     */
+    get showError(): boolean {
+      return this.___checkForError();
+    }
+
+    /**
+     * The validation message shown by a given Validator object. If the control
+     * is in a valid state this should be falsy.
+     */
+    get validationMessage(): string {
+      return this.internals.validationMessage;
+    }
+
+    /**
+     * The element's validity state after evaluating the control's Validators.
+     * This property implements the same patterns as the built-in constraint
+     * validation strategy.
+     */
+    get validity(): ValidityState {
+      return this.internals.validity;
+    }
+
+    /**
+     * Exists to control when an error should be displayed
+     * @private
+     */
+     ___forceError = false;
+
+    /**
+     * Is set in the constructor, will resolve whenever the validationTarget
+     * is ready so `internals.setValidity` can have access to the element
+     * and will not throw.
+     * @private
+     */
+    ___ready: Promise<void>;
+
+    constructor(...args: any[]) {
+      super(...args);
+      this.addEventListener('focus', this.___onFocus);
+      this.addEventListener('blur', this.___onBlur);
+      this.addEventListener('invalid', this.___onInvalid);
+
+      let tick = 0;
+      this.___ready = new Promise((resolve, reject) => {
+        const id = setInterval(() => {
+          if (tick >= 100) {
```

Comment:
I'm not seeing where `tick` gets incremented in a way that would ever meet this condition...

Reply:
Yeah, looks like that got missed.